packages/Popover/src/Popover.js

```diff
@@ -296,11 +302,66 @@ class Popover extends React.Component {
     }
   };
 
+  handleKeyDown = event => {
+    if (event.key === "Tab" && this.isOpen() && this.$trigger) {
+      const isFocusOnFirst = this.focusIsOnCertainElementInPopover("first") || document.activeElement === this.$content;
+      const isFocusOnLast = this.focusIsOnCertainElementInPopover("last");
+
+      if (event.shiftKey && isFocusOnFirst) {
+        event.preventDefault();
+        this.restoreTabIndexes();
+        this.focusableElements[this.triggerFocusIndex].focus();
+      } else if (!event.shiftKey && isFocusOnLast) {
+        event.preventDefault();
+        this.restoreTabIndexes();
+        this.focusableElements[this.triggerFocusIndex + 1].focus();
+      }
+    }
+  };
+
+  focusIsOnCertainElementInPopover = which => {
+    const focusableElementsInPopover = this.$content.querySelectorAll(focusableElementSelector);
+    const index = which === "first" ? 0 : focusableElementsInPopover.length - 1;
+
+    return document.activeElement === focusableElementsInPopover[index];
+  };
+
+  elementIsDescendentOfPopover = elem => {
+    let elementsParent = elem.parentNode;
+    while (elementsParent !== null) {
+      if (elementsParent.getAttribute && elementsParent.getAttribute("data-pka-anchor") === "popover.content") {
+        return true;
+      }
+      elementsParent = elementsParent.parentNode;
+    }
+    return false;
+  };
+
+  restoreTabIndexes = () => {
+    this.focusableElements.forEach(focusableElement => {
+      if (focusableElement !== this.$trigger && !this.elementIsDescendentOfPopover(focusableElement)) {
+        focusableElement.tabIndex = this.originalTabIndexes.reverse().pop(); // eslint-disable-line no-param-reassign
```

Comment:
`reverse()` is in-place, and this is done in a loop, so probably there's an issue with reversing multiple times. `this.originalTabIndexes.shift()` should work?